src/logic/updates.ts

```diff
@@ -39,31 +44,28 @@ function handleUpdate<T extends keyof SubscriptionEventsEmitter>(handler: Update
   }
 }
 
-export function friendshipUpdateHandler(getContext: () => SharedContext, logger: ILogger) {
+export function friendshipUpdateHandler(rpcContext: ISubscribersContext, logger: ILogger) {
   return handleUpdate<'friendshipUpdate'>((update) => {
-    const context = getContext()
-    const updateEmitter = context.subscribers[update.to]
+    const updateEmitter = rpcContext.getSubscriber(update.to)
     if (updateEmitter) {
       updateEmitter.emit('friendshipUpdate', update)
     }
   }, logger)
 }
 
 export function friendConnectivityUpdateHandler(
-  getContext: () => SharedContext,
+  rpcContext: ISubscribersContext,
   logger: ILogger,
   db: IDatabaseComponent
 ) {
   return handleUpdate<'friendConnectivityUpdate'>(async (update) => {
-    const context = getContext()
-
-    const onlineSubscribers = Object.keys(context.subscribers)
+    const onlineSubscribers = rpcContext.getSubscribersAddresses()
     const friends = await db.getOnlineFriends(update.address, onlineSubscribers)
 
     friends.forEach(({ address: friendAddress }) => {
       const normalizedAddress = normalizeAddress(friendAddress)
 
-      const emitter = context.subscribers[normalizedAddress]
+      const emitter = rpcContext.getSubscriber(normalizedAddress)
```

Comment:
Also maybe we can add it on rpcContext if it also needed there (still talking about address normalization)